lib/retriable.rb

```diff
@@ -6,14 +6,22 @@
 module Retriable
   module_function
 
-  def self.configure
+  def configure
     yield(config)
   end
 
   def config
     @config ||= Config.new
   end
 
+  def retriable_with_context(context_key, options = {}, &block)
```

Comment:
> I didn't like the idea of having it be `#retriable` map to `Retriable.retriable` but `#retriable_with_context` in Kernel map to `Retriable.with_context`.

i tend to view adding methods to `Kernel` as kind of way more intrusive than, for instance, implementing `method_missing` within the gem... i would probably vote to deprecate it altogether if i had my druthers (which i don't; just putting that out there)

point being, i think `Retriable.with_context` reads really well, and if the kernel extension can't match the semantics, it doesn't really bother me (though i agree it's not quite as neat & tidy)